backend/tests/test_GP_2.py

```diff
@@ -0,0 +1,116 @@
+import base64
+import unittest
+import json
+from model.users import UsersModel
+from tests.base_test import BaseTest
+
+
+class UnitTestOfUS(BaseTest):
+    def basic_setup(self):
+        self.user = UsersModel("test", "test@gmail.com")
+        self.user.hash_password("test")
+        self.user.save_to_db()
+
+        res = self.client.post("/login", data={"email": self.user.email, "password": "test"})
+        self.token = json.loads(res.data)["token"]
+
+    # TEST TASK 2
+    def test_modify_username(self):
+        with self.app.app_context():
+            self.basic_setup()
+
+            data_new = {
+                'username': 'francina',
+                'password': 'test'
+            }
+            res = self.client.put(f"/user/{self.user.email}", data=data_new, headers={
+                "Authorization": 'Basic ' + base64.b64encode((self.token + ":").encode('ascii')).decode('ascii')
+            })
+            self.assertEqual(200, res.status_code)
+            self.assertEqual((json.loads(res.data)["user"])['username'], data_new['username'])
+
+    def test_modify_email(self):
+        with self.app.app_context():
+            self.basic_setup()
+            data_new = {
+                'email': 'francina@gmail.com',
+                'password': 'test'
+            }
+            res = self.client.put(f"/user/{self.user.email}", data=data_new, headers={
+                "Authorization": 'Basic ' + base64.b64encode((self.token + ":").encode('ascii')).decode('ascii')
+            })
+            self.assertEqual(200, res.status_code)
+            self.assertEqual((json.loads(res.data)["user"])['email'], data_new['email'])
```

Comment:
Mateix comentari que el anterior.
```suggestion
            self.assertEqual(data_new['email'], (json.loads(res.data)["user"])['email'])
```